eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -156,58 +154,40 @@ package object eclair {
   }
 
   /**
-    * We use this in the context of timestamp filtering, when we don't need an upper bound.
-    */
+   * We use this in the context of timestamp filtering, when we don't need an upper bound.
+   */
   val MaxEpochSeconds = Duration.fromNanos(Long.MaxValue).toSeconds
 
-  /**
-    * One MilliSatoshi is a thousand of a Satoshi, the smallest unit usable in bitcoin
-    * @param amount
-    */
-  case class MilliSatoshi(amount: Long) {
+  implicit class LongToBtcAmount(l: Long) {
     // @formatter:off
-    def toLong = amount
-    def +(other: MilliSatoshi) = MilliSatoshi(amount + other.amount)
-    def -(other: MilliSatoshi) = MilliSatoshi(amount - other.amount)
-    def *(m: Long) = MilliSatoshi(amount * m)
-    def *(m: Double) = MilliSatoshi((amount * m).toLong)
-    def /(d: Long) = MilliSatoshi(amount / d)
-    def compare(other: MilliSatoshi): Int = if (amount == other.amount) 0 else if (amount < other.amount) -1 else 1
-    def <= (that: MilliSatoshi): Boolean = compare(that) <= 0
-    def >= (that: MilliSatoshi): Boolean = compare(that) >= 0
-    def <  (that: MilliSatoshi): Boolean = compare(that) < 0
-    def >  (that: MilliSatoshi): Boolean = compare(that) > 0
-    def unary_-() = MilliSatoshi(-amount)
-    def truncateToSatoshi: Satoshi = Satoshi(amount / 1000)
+    def msat: MilliSatoshi = MilliSatoshi(l)
+    def sat: Satoshi = Satoshi(l)
+    def mbtc: MilliBtc = MilliBtc(l)
+    def btc: Btc = Btc(l)
     // @formatter:on
   }
 
-  implicit class ToMilliSatoshiConversion(amount: BtcAmount) {
-    def toMilliSatoshi: MilliSatoshi = amount match {
-      case sat: Satoshi => satoshi2millisatoshi(sat)
-      case millis: MilliBtc => satoshi2millisatoshi(millibtc2satoshi(millis))
-      case bitcoin: Btc => satoshi2millisatoshi(btc2satoshi(bitcoin))
-    }
-  }
-
   implicit object NumericMilliSatoshi extends Numeric[MilliSatoshi] {
-    override def plus(x: MilliSatoshi, y: MilliSatoshi): MilliSatoshi = MilliSatoshi(x.amount + y.amount)
-    override def minus(x: MilliSatoshi, y: MilliSatoshi): MilliSatoshi = MilliSatoshi(x.amount - y.amount)
-    override def times(x: MilliSatoshi, y: MilliSatoshi): MilliSatoshi = MilliSatoshi(x.amount * y.amount)
-    override def negate(x: MilliSatoshi): MilliSatoshi = MilliSatoshi(-x.amount)
+    // @formatter:off
+    override def plus(x: MilliSatoshi, y: MilliSatoshi): MilliSatoshi = x + y
```

Comment:
A comment explaining why we also need to implement `Numeric[MilliSatoshi]` would be helpful.